pkg/controller/queuejob/utils_test.go

```diff
@@ -0,0 +1,236 @@
+package queuejob
+
+import (
+	"testing"
+
+	"github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	arbv1 "github.com/project-codeflare/multi-cluster-app-dispatcher/pkg/apis/controller/v1beta1"
+)
+
+
+func TestGetXQJFullName(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+
+	tests := []struct {
+		name      string
+		qj        *arbv1.AppWrapper
+		expected  string
+	}{
+		{
+			name: "valid qj",
+			qj: &arbv1.AppWrapper{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "qjName",
+					Namespace: "qjNamespace",
+				},
+			},
+			expected: "qjName_qjNamespace",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := GetXQJFullName(tt.qj)
+			g.Expect(result).To(gomega.Equal(tt.expected))
+		})
+	}
+}
+
+
+func TestHigherSystemPriorityQJ(t *testing.T) { 
+	g := gomega.NewGomegaWithT(t)
+
+	tests := []struct {
+		name     string
+		qj1      *arbv1.AppWrapper
+		qj2      *arbv1.AppWrapper
+		expected bool
+	}{
+		{
+			name: "lower priority qj1",
+			qj1: &arbv1.AppWrapper{
+				Status: arbv1.AppWrapperStatus{
+					SystemPriority: 1,
+				},
+			},
+			qj2: &arbv1.AppWrapper{
+				Status: arbv1.AppWrapperStatus{
+					SystemPriority: 2,
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "higher priority qj1",
+			qj1: &arbv1.AppWrapper{
+				Status: arbv1.AppWrapperStatus{
+					SystemPriority: 2,
+				},
+			},
+			qj2: &arbv1.AppWrapper{
+				Status: arbv1.AppWrapperStatus{
+					SystemPriority: 1,
+				},
+			},
+			expected: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := HigherSystemPriorityQJ(tt.qj1, tt.qj2)
```

Comment:
Can you change the method signature to not use interface{} and use AppWrapper ?

Reply:
I added a new commit to make use of `AppWrapper`, it required me to change the heap.go functions as well to use the AppWrapper type. I'm not sure if this is the ideal way of doing it, hoping to get your thoughts/advice on the changes made. Thank you @z103cb

Reply:
They are LGTM. @astefanutti please review this change and approve it. I think it might be conflict with your k8s upgrade PR>